process/epochStart/epoch_start_processor.go

```diff
@@ -0,0 +1,64 @@
+package epochStart
+
+import (
+	"encoding/hex"
+
+	"github.com/ElrondNetwork/covalent-indexer-go/process/utility"
+	"github.com/ElrondNetwork/covalent-indexer-go/schemaV2"
+	"github.com/ElrondNetwork/elrond-go-core/data/api"
+)
+
+type epochStartInfoProcessor struct {
+}
+
+// NewEpochStartInfoProcessor will create a new instance of epoch start info processor
+func NewEpochStartInfoProcessor() *epochStartInfoProcessor {
+	return &epochStartInfoProcessor{}
+}
+
+// ProcessEpochStartInfo converts receipts api epoch start info to a specific structure defined by avro schema
+func (esi *epochStartInfoProcessor) ProcessEpochStartInfo(apiEpochInfo *api.EpochStartInfo) (*schemaV2.EpochStartInfo, error) {
+	if apiEpochInfo == nil {
+		return nil, nil
```

Comment:
Why not return nil, ErrNilEpochStartInfo and threat this error in the caller side however you want. Returning nil, nil is dangerous.

Reply:
Yep, the only problem is that this method si exported in the interface so this is public, therefore anyone can use it. If it will be used and the normal expectation is that if err is nil then the returned type will be not, this could trigger some panic situations. You can have the same result if you will change in the caller side, I guess that for now is just one place in process/hyper_block_processor.go -> L49, and in L50 you will use this approach:

`if err != nil && !errors.Is(err, process. ErrNilEpochStartInfo)`